src/agent.ts

```diff
@@ -378,6 +378,11 @@ export class Agent {
       // JWTs
       await interxn.processInteractionToken(token)
       await this.storage.store.interactionToken(token)
+    } else if (interxn.lastMessage.encode() === jwt) {
```

Comment:
please avoid this encode() step happening twice, it's kinda expensive(ish) and unnecessary, it happens in the previous check as well